instrumentation/servlet/servlet-3.1/src/main/java/io/opentelemetry/instrumentation/hypertrace/servlet/v3_1/Servlet31Advice.java

```diff
@@ -97,7 +97,14 @@ public static Object start(
   public static void stopSpan(
       @Advice.Argument(0) ServletRequest request,
       @Advice.Argument(1) ServletResponse response,
+      @Advice.Thrown Throwable throwable,
       @Advice.Local("rootStart") Boolean rootStart) {
+
+    if (throwable instanceof ClassCastException) {
+      ClassCastException cast = (ClassCastException) throwable;
+      System.out.println(cast.getMessage().contains("hypertrace"));
```

Comment:
Worth to mention why this piece is important? Maybe abstracting `"hypertrace"` into something meaningful like `HYPERTRACE_OWN_PREFIX` or something that makes sense?

Reply:
ignore this code completely this is just a nasty example.

The right solution could parse the class name and store it into the global config. The global config would be used by instrumentation points that create wrappers. If no exception is recorded for the wrapped type the wrapping would be applied.

Reply:
Another interesting fact in this example is the place where the exceptions are caught. In servlet this can be done in exit filter advice. In other frameworks it can be a different place.

The problem also is if the casting would be inside user defined try-catch(Throwable) block, this instrumentation would not catch this. I have tried to define adice for `ClassCastException` but that does not seem to be triggered.